### PR TITLE
B-21488: Fixing UB Warning Message Display

### PR DESCRIPTION
**Pull request by TevinAdams**
## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21488)
##[I-13640](https://www13.v1host.com/USTRANSCOM38/Issue.mvc/Summary?oidToken=Issue%3A1044154)

## Summary
This is a fix for the work pushed up for B-21488.  Forgot to add check for UB Shipments for the UB Excess weight logic.

The UB shipment excess weight warning was showing up on moves that had no UB shipments. This was because there was no condition added to stop the UB logic from running when there is no UB Shipments. Simple fix. Add a check for UB shipments where if there are UB shipments then run the logic if not skip it.

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the milmove and create a move without a UB shipment
2. Counsel the move
3. Approve the move as TOO
4. As a Prime sim update the estimated weight with a weight that will trigger an excess weight for the non UB shipment
5. Go view the Move as a TOO and you should not see the UB Excess weight message because there is no UB shipments

OPTIONAL(This was tested in the last PR and nothing changed the implementation)
7. Create a move with a UB shipment.
8. Update the estimated weight on the UB shipment to where it triggers and access weight
9. View the move as TOO. Should see the UB Shipment excess weight message

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

File: pkg/services/move/move_weights.go [hunk 1]
```diff
@@ -136,12 +136,19 @@ func saveMoveExcessWeightValues(appCtx appcontext.AppContext, move *models.Move,
...
-	if (int(float32(ubWeightAllowance)*RiskOfExcessThreshold) <= sumOfWeights.SumEstimatedWeightOfUbShipments) || (int(float32(ubWeightAllowance)*RiskOfExcessThreshold) <= sumOfWeights.SumActualWeightOfUbShipments) {
+	var hasUbShipments bool
+	for _, shipment := range move.MTOShipments {
+		if shipment.ShipmentType == models.MTOShipmentTypeUnaccompaniedBaggage {
+			hasUbShipments = true
+		}
+	}
+
+	// Check for risk of excess of UB allowance if there are UB shipments
+	if hasUbShipments && ((int(float32(ubWeightAllowance)*RiskOfExcessThreshold) <= sumOfWeights.SumEstimatedWeightOfUbShipments) || (int(float32(ubWeightAllowance)*RiskOfExcessThreshold) <= sumOfWeights.SumActualWeightOfUbShipments)) {
```
Comment (danieljordan-caci) [inline pkg/services/move/move_weights.go:147]: For readability, can you break this down like so?
```
threshold := int(float32(ubWeightAllowance) * RiskOfExcessThreshold)

isWeightExceedingThreshold := (threshold <= sumOfWeights.SumEstimatedWeightOfUbShipments) || (threshold <= sumOfWeights.SumActualWeightOfUbShipments)

if hasUbShipments && isWeightExceedingThreshold {
}
```

Comment (danieljordan-caci) [reply pkg/services/move/move_weights.go:147]: It's just kind of an ugly mess to look at when it's one big ole line haha

Comment (TevinAdams) [reply pkg/services/move/move_weights.go]: @danieljordan-caci  Done

**Comment by TevinAdams** [comment]
> Didn't we agree to add a breakdown of UB stuff here? I remember that being a thing. @transcom/mountain-movers
> 
> Screenshot 2024-12-16 at 11 10 01 AM
> 
> Otherwise I'm able to see both alerts at the same time, does not show when there isn't a UB shipment. Screenshot 2024-12-16 at 11 12 17 AM
> 
> Any thoughts on also adding a UI check to display the UB weight alert is the UB flag is on and the move contains UB shipments?

@traskowskycaci I believe someone else had a story for that. I remember at PI planning Maria showed me and Cameron a box with the UB allowance. But someone else did that.

**Comment by ajlusk** [comment]
Not a review since I don't know the details about all the other stuff but the scenario in which I encountered this issue appears to be resolved.

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
In addition to my outstanding questions, it looks like the action count is off.
This should say `2` in the action flags, but only says `1`
Screenshot 2024-12-16 at 11 31 19 AM

Also when I dismissed the non-UB alert, the move status changed to `APPROVED`, but I still had the UB one present.
move is approved
Screenshot 2024-12-16 at 11 33 03 AM
still have an action
Screenshot 2024-12-16 at 11 33 11 AM

**Comment by TevinAdams** [comment]
> > > Didn't we agree to add a breakdown of UB stuff here? I remember that being a thing. @transcom/mountain-movers
> > > Screenshot 2024-12-16 at 11 10 01 AM
> > > Otherwise I'm able to see both alerts at the same time, does not show when there isn't a UB shipment. Screenshot 2024-12-16 at 11 12 17 AM
> > > Any thoughts on also adding a UI check to display the UB weight alert is the UB flag is on and the move contains UB shipments?
> > 
> > 
> > @traskowskycaci I believe someone else had a story for that. I remember at PI planning Maria showed me and Cameron a box with the UB allowance. But someone else did that. There was a box specific to UB from what I saw.
> 
> There is not an additional BL that I'm seeing that is covering this work.
Its fine ill add something similar to what i saw. A box to the right with UB allowance information?

**Comment by danieljordan-caci** [comment]
> > > > Didn't we agree to add a breakdown of UB stuff here? I remember that being a thing. @transcom/mountain-movers
> > > > Screenshot 2024-12-16 at 11 10 01 AM
> > > > Otherwise I'm able to see both alerts at the same time, does not show when there isn't a UB shipment. Screenshot 2024-12-16 at 11 12 17 AM
> > > > Any thoughts on also adding a UI check to display the UB weight alert is the UB flag is on and the move contains UB shipments?
> > > 
> > > 
> > > @traskowskycaci I believe someone else had a story for that. I remember at PI planning Maria showed me and Cameron a box with the UB allowance. But someone else did that. There was a box specific to UB from what I saw.
> > 
> > 
> > There is not an additional BL that I'm seeing that is covering this work.
> > Its fine ill add something similar to what i saw. A box to the right with UB allowance information?

No, you need to append to what's already there - underneath the "Hide Breakdown" section.

**Comment by TevinAdams** [comment]
> > > Didn't we agree to add a breakdown of UB stuff here? I remember that being a thing. @transcom/mountain-movers
> > > Screenshot 2024-12-16 at 11 10 01 AM
> > > Otherwise I'm able to see both alerts at the same time, does not show when there isn't a UB shipment. Screenshot 2024-12-16 at 11 12 17 AM
> > > Any thoughts on also adding a UI check to display the UB weight alert is the UB flag is on and the move contains UB shipments?
> > 
> > 
> > @traskowskycaci I believe someone else had a story for that. I remember at PI planning Maria showed me and Cameron a box with the UB allowance. But someone else did that. There was a box specific to UB from what I saw.
> 
> There is not an additional BL that I'm seeing that is covering this work.

@danieljordan-caci Breakdown added
image

**Comment by TevinAdams** [comment]
> Member

@danieljordan-caci Excess Risk count has been fixed
image

File: src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx [hunk 2]
```diff
@@ -976,6 +978,10 @@ export const MoveTaskOrder = (props) => {
           ? formatWeight(Math.round(estimatedNTSReleaseWeightTotal * 1.1))
           : '—'}
       </div>
+      <div>110% Estimated UB</div>
+      <div className={moveTaskOrderStyles.subValue}>
+        {Number.isFinite(estimatedUBWeightTotal) ? formatWeight(Math.round(estimatedUBWeightTotal * 1.1)) : '—'}
+      </div>
```
Comment (danieljordan-caci) [inline src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx:984]: Need a test

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]

File: src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx (hunk 2 above)
Comment (TevinAdams) [reply src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx]: @danieljordan-caci  test added

**Review by traskowskycaci — COMMENTED** [review]
Huddled with Tevin for a demo of the UI here, had a few questions on whether we want to allow multiple excess weight acknowledgements to happen, and whether we want to keep the move in approvals requested until all excess weight risks are acknowledged, when there are multiple, one for the move and one for UBs

**Review by traskowskycaci — APPROVED** [review]
Would be good to get this tested and in front of the PO, and see if the questions I had arise and discuss how we'd like to handle that work

**Review by danieljordan-caci — APPROVED** [review]

**Review by danieljordan-caci — APPROVED** [review]

**Review by paulstonebraker — APPROVED** [review]

**Pull request merged**